### DISCUSSION TITLE
WinUI3 mask edit.  Anyone has a solution that is user friendly

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#17094] [General]
Our application is an enterprise forms application, which there are many mask edit fields needed for input.  Currently the existing field input in WinUI3 is TextBox or NumericTextBox.  Both have no real support for mask edit.

The communitytoolkit version 8 does have an extension helper for TextBox called TextBoxExtensions that allows the use of a mask such as 99/99/9999 but this is not user friendly since you must always enter the date as 01/23/1980 whereas our users would like to just enter 1/23/1980.  The same thing holds true for IP addresses and other types of entry fields.

The NumericTextBox also has limitations such as allowing for the money symbol before or after the input value, say the user enters $100.00 and then enters, the control sets the value to 0.  This happens in most copy/paste use cases.

Anyone know of another solution other than using 3rd party tools from Maui?  If I go this route I can't u build the web version of the system.

Any pointers welcome.

**Comment by nickrandolph** [comment]
@cconner100 one option would be to clone the TextBoxExtensions and customize it to the behavior that you want (for example you could adapt the mask to indicate the acceptable number of characters. For example you could have "9[1-2]" meaning that 9 needs to occur at least 1 times and no more than 2 times.

For NumericTextBox I would also suggest adapting the TextBoxExtensions and apply them to the NumericTextBox.

Were there specific 3rd party tools that are available for Maui that provide the requirements you're after?

**Reply by cconner100** [reply]
We have used syncfusion in our previous xamarin.forms application which seems to do what we needed

[https://help.syncfusion.com/maui/masked-entry/getting-started](url)
and
[https://help.syncfusion.com/maui/numericentry/formatting](url)
